scrabble.js

```diff
@@ -1,8 +1,307 @@
-var Scrabble = function() {};
+// scoreChart[1] ==>>> [ 'A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T' ]
+// Object.keys(this.scoreChart) ====>>> [ '1', '2', '3', '4', '5', '8', '10' ]
+var Scrabble = {
+  scoreChart: {
+    1: ["A","E","I","O","U","L","N","R","S","T"],
+    2: ["D","G"],
+    3: ["B","C","M","P"],
+    4: ["F","H","V","W","Y"],
+    5:["K"],
+    8: ["J","X"],
+    10: ["Q","Z"]
+  },
 
-// YOUR CODE HERE
-Scrabble.prototype.helloWorld = function() {
-  return 'hello world!';
+  letterDistribution: {
+      A: 9, B: 2, C: 2, D: 4, E: 12, F: 2, G: 3, H: 2, I: 9, J: 1, K: 1, L: 4, M: 2, N: 6, O: 8, P: 2, Q: 1, R: 6, S: 4, T: 6, U: 4, V: 2, W: 2, X: 1, Y: 2, Z: 1},
+
+  // score(word): returns the total score value for the given word. The word is input as a  string (case insensitive).
+  score: function(wordPlayed) {
+    var score = 0;
+    var word = wordPlayed.toUpperCase();
+    var wordArray = word.split("");
+    var scoreValues = Object.keys(this.scoreChart);
+    var scoreLetterValue = this.scoreChart;
+
+    wordArray.forEach(function(letter) {
+      scoreValues.forEach(function(value) {
```

Comment:
This solution is using a lot of unnecessary iteration over every score value - how do you think you might optimize?